[common] Make dolphinscheduler_env.sh work when start server

zhongjiajie opened the pull request:
* Change dist tarball `dolphinscheduler_env.sh` location
  from `bin/` to `conf/`, which users could finish their
  change configuration operation in one single directory.
  and we only need to add `$DOLPHINSCHEDULER_HOME/conf`
  when we start our sever instead of adding both
  `$DOLPHINSCHEDULER_HOME/conf` and `$DOLPHINSCHEDULER_HOME/bin`
* Change the `start.sh`'s path of `dolphinscheduler_env.sh`
* Change the setting order of `dolphinscheduler_env.sh`
* Change the related docs

close: #9727

zhongjiajie committed:
[common] Make dolphinscheduler_env.sh work

* Change dist tarball `dolphinscheduler_env.sh` location
  from `bin/` to `conf/`, which users could finish their
  change configuration operation in one single directory.
  and we only need to add `$DOLPHINSCHEDULER_HOME/conf`
  when we start our sever instead of adding both
  `$DOLPHINSCHEDULER_HOME/conf` and `$DOLPHINSCHEDULER_HOME/bin`
* Change the `start.sh`'s path of `dolphinscheduler_env.sh`
* Change the setting order of `dolphinscheduler_env.sh`
* Change the related docs

kezhenxu94 commented on dolphinscheduler-master/src/main/bin/start.sh:
```diff
@@ -19,7 +19,7 @@
 DOLPHINSCHEDULER_HOME=${DOLPHINSCHEDULER_HOME:-$(cd $BIN_DIR/..; pwd)}
 
-source "$BIN_DIR/dolphinscheduler_env.sh"
+source "$BIN_DIR/../conf/dolphinscheduler_env.sh"
```
What about just 

```suggestion
source "$DOLPHINSCHEDULER_HOME/conf/dolphinscheduler_env.sh"
```

zhongjiajie replied:
Will change it

zhongjiajie replied:
Done and could you please take a look again? BTW, why we should add file `dolphinscheduler_env.sh` in all servers instead of using the file in the path `bin/env/dolphinscheduler_env.sh`?

kezhenxu94 replied:
> BTW, why we should add file `dolphinscheduler_env.sh` in all servers instead of using the file in the path `bin/env/dolphinscheduler_env.sh`?

See my comment https://github.com/apache/dolphinscheduler/issues/9675#issuecomment-1107996038 . The short answer is that `bin/env/dolphinscheduler_env.sh` is only available when you use `install.sh` to install all components in a single machine. If you install each component in a separate machine, `bin/env/dolphinscheduler_env.sh` is not available there, also cc @SbloodyS

kezhenxu94 reviewed (dismissed):
LGTM overall

zhongjiajie committed:
Change related path for `dolphinscheduler_env.sh`

SbloodyS commented:
I have a suggestion in #9675 . Because I found that many PR and issues are related to this problem. If everyone agrees, I think we can add it to this PR to avoid duplication of work. Then close other related issues.

Please task a look @zhongjiajie @kezhenxu94 @caishunfeng

zhongjiajie committed:
dolphinscheduler_env.sh overwrite when start with shell in bin dir

`bin/env/dolphinscheduler_env.sh` will overwrite the `<server>/conf/dolphinscheduler_env.sh`
when start the server using `bin/dolphinsceduler_daemon.sh` or `bin/install.sh`

zhongjiajie committed:
Merge remote-tracking branch 'upstream/dev' into b-env-not-work

zhongjiajie commented:
I add feature we metiond in https://github.com/apache/dolphinscheduler/issues/9675#issuecomment-1108041268, and after this patch merged, user s who using commad `./bin/dolphinscheduler_daemon.sh start <server>` or `./bin/install.sh` to start dolphinscheduler, the server env config will be overwrite by `./bin/env/dolphinscheudler_env.sh`.

zhongjiajie commented:
I going to merge this one
